### PR TITLE
Set Azure OpenAI API version in global configuration

### PR DESCRIPTION
**Pull request by sogue**
## Purpose

Set Azure OpenAI API version in global configuration
[Older versions retire on 2 April 2024. ](https://learn.microsoft.com/en-us/azure/ai-services/openai/reference)

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[x] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[x] No
```

## Type of change

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](../CONTRIBUTING.md#submit-pr) for more details.

- [x] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works
- [ ] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ ] I ran `python -m mypy` to check for type errors
- [ ] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

**Comment by sogue** [comment]
@microsoft-github-policy-service agree

File: app/backend/app.py
Comment (pamelafox) [inline app/backend/app.py:311]: This will cause a key error in old environments, please use get() and provide a default value.

**Review by pamelafox — CHANGES_REQUESTED** [review]
I think an env variable is a good idea, but we cant assume it exists.

File: app/backend/app.py
Comment (pamelafox) [reply app/backend/app.py:311]: The latest. I just need to make sure it works with gpt-4-vision but I'll test that before merging.

File: app/backend/app.py
Comment (pamelafox) [inline app/backend/app.py:311]:
```suggestion
        api_version = os.getenv("AZURE_OPENAI_API_VERSION", "2024-03-01-preview")
```

slightly shorter wrapper func

File: infra/main.bicep
Comment (pamelafox) [inline infra/main.bicep:45]: Could you also add to outputs at the bottom, just in case someone sets it by changing line 45? (vs using `azd env set`)

File: app/backend/app.py
Comment (pamelafox) [inline app/backend/app.py:311]: Sorry, I just tested in another repo, and we actually need this:

```suggestion
        api_version = os.getenv("AZURE_OPENAI_API_VERSION") or "2024-02-15-preview"
```

To account for that variable being an empty string (versus None)

**Review by pamelafox — APPROVED** [review]
Just tested with normal gpt-35 requests and gpt-4-vision requests in an existing environment, new version seems to work fine. Will merge after my deploy is successful as well.

**Pull request merged**
